### PR TITLE
Bump min sdk to 2.6.0 so we can use extension methods.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
@jacob314 @DanTup is the server bump necessary here?

### CODE REVIEW

**Comment by DanTup** [comment]
> is the server bump necessary here?

I'm not sure if it's necessary, but I think it probably makes sense to keep them in sync.

When are we next releasing? If it's before v2.6 hits the Flutter stable channel, then Flutter users will find they can't upgrade to the latest version when the release notes go out (which might make them sad if they wanted a new feature).

Similarly, will this affect anyone that depends on devtools (I think webdev/dwds/flutter)?

File: packages/devtools/pubspec.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ author: Dart Team <misc@dartlang.org>
 homepage: https://github.com/flutter/devtools
 
 environment:
-  sdk: '>=2.3.0 <3.0.0'
+  sdk: '>=2.6.0 <3.0.0'
```
Comment (jacob314) [inline packages/devtools/pubspec.yaml]: we should keep this version and the version in the server at 2.3 or maybe 2.5 unless there is a strong reason not to. That way people with slightly old versions of Flutter can still use the latest version of DevTools. By contrast, we precompile devtools_app so using a tip of trunk sdk does not impact our users.

File: packages/devtools_server/pubspec.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ author: Dart Team <misc@dartlang.org>
 homepage: https://github.com/flutter/devtools
 
 environment:
-  sdk: '>=2.3.0 <3.0.0'
+  sdk: '>=2.6.0 <3.0.0'
```
Comment (jacob314) [inline packages/devtools_server/pubspec.yaml]: move this back to 2.3 unless we really really need extension methods for the server package.

File: packages/devtools/pubspec.yaml (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools/pubspec.yaml]: Done. Now the only bump in this CL is to `devtools_app/pubspec.yaml`

File: packages/devtools_server/pubspec.yaml (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_server/pubspec.yaml]: done

File: packages/devtools/pubspec.yaml (hunk 1 above)
Comment (jacob314) [reply packages/devtools/pubspec.yaml]: maybe add a comment in the other pubspec.yaml explaining why we keep intentionally diverge the sdk versions?

**Comment by jacob314** [comment]
lgtm

**Pull request merged**